spec/forms/respondent_form_spec.rb

```diff
@@ -154,7 +154,7 @@
     before { allow(respondent_form.target).to receive :enqueue_fee_group_reference_request }
 
     it 'addresses reloaded on save' do
-      expect(respondent_form.target.addresses).to receive(:reload)
+      expect_any_instance_of(respondent_form.target.addresses.class).to receive(:reload)
```

Comment:
We cannot assume that respondent_form.target.addresses is going to be the same object all the time - in rails 5 it isn't - we get a new proxy object each time.
We should not really be mocking interfaces that are not ours, but the old test did, so i have simply made this test work - not ideal using expect_any_instance_of, but without re designing the interfaces, this was the best I could achieve